lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -33,7 +34,10 @@ class ConnectionFailure < StandardError; end
     config_param :scheme, :string, :default => 'http'
     config_param :hosts, :string, :default => nil
     config_param :target_index_key, :string, :default => nil
-    config_param :target_type_key, :string, :default => nil
+    config_param :target_type_key, :string, :default => nil,
+                 :deprecated => <<EOC
+Elasticsearch 7.x or above will ignore this config. Please use fixed type_name instead.
+EOC
```

Comment:
How about using this ruby 2.3 functionality to remove the leading whitespace? (depends on your min ruby version though)

```ruby
def test
  <<~END
    First content line.
      Two spaces here.
    No space here.
  END
end

test
# => "First content line.\n  Two spaces here.\nNo space here.\n"
```

Source:
https://stackoverflow.com/a/34888238/118007

Reply:
I wish I could.
But this plugin should support Ruby 2.1 and Ruby 2.2.
I want to leave as-is this nor now. :confounded: